### PR TITLE
feat: add attr "Dataframe._record_key" to store the Graviti __record_key

### PR DESCRIPTION
**Pull request by linjiX**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-2fencx7 在 DataFrame 中接入 __record_key](https://app.clickup.com/t/2fencx7)

File: graviti/manager/sheets.py [hunk 1]
```diff
@@ -55,13 +57,18 @@ def _init_data(self) -> None:
...
             schema = PortexType.from_yaml(sheet["schema"])
 
+            patype = schema.to_pyarrow()
+
             factory = LazyFactory(
                 sheet["record_count"],
                 128,
                 partial(self._list_data, sheet_name=sheet_name),
-                schema.to_pyarrow(),
+                pa.struct([pa.field(RECORD_KEY, pa.string()), *patype]),
```
Comment (graczhual) [inline graviti/manager/sheets.py:66]: do we need to do this change in search function?

Comment (linjiX) [reply graviti/manager/sheets.py:66]: do we need `__record_key` in search result?

Comment (graczhual) [reply graviti/manager/sheets.py:66]: There is no need for it

File: graviti/manager/sheets.py [hunk 2]
```diff
@@ -55,13 +57,18 @@ def _init_data(self) -> None:
...
             )
+
             paging_lists = factory.create_lists(schema.get_keys())
+            paging_lists[RECORD_KEY] = factory.create_list((RECORD_KEY,))
```
Comment (Lee-000) [inline graviti/manager/sheets.py:70]: Why not add `(RECORD_KEY,)` to the keys and save the `create_list` call?

**Review by graczhual — APPROVED** [review]

**Review by AChenQ — APPROVED** [review]

File: graviti/manager/sheets.py (hunk 2 above)
Comment (linjiX) [reply graviti/manager/sheets.py:70]: `create_lists` also calls `create_list`, is there any difference?

**Review by Lee-000 — APPROVED** [review]

**Pull request merged**